### PR TITLE
(#115) discord.js v14

### LINKED ISSUE #115: Support Discord.JS v14

### PR DESCRIPTION
**Pull request by mkevenaar**
Fixes #115

### CODE REVIEW

File: src/commands/general/guildstats.js [hunk 1]
```diff
@@ -25,27 +23,29 @@ export async function execute(interaction, client) {
...
-      .size;
+    let catCount = await interaction.guild.channels.cache.filter(
+      (x) => x.type === ChannelType.GuildCategory
+    ).size;
     // Get the amount of role
     let roleCount = await interaction.guild.roles.cache.size;
 
     // Get server verification level
-    let verifyLevel = await interaction.guild.verificationLevel.toLowerCase();
+    let verifyLevel = await interaction.guild.verificationLevel.toString().toLowerCase();
```
Comment (davidzwa) [inline src/commands/general/guildstats.js:40]: Just wanted to let you know that this might break if the level is undefined. Up to you to decide

**Review by Lucxjo — APPROVED** [review]
All looks good! I don't believe that anything needs changing at this time

File: src/database/models/guild.js [hunk 2]
```diff
@@ -1,9 +1,15 @@
 import mongoose from 'mongoose';
 
 const GuildSchema = new mongoose.Schema({
-  id: { type: String }, //ID of the guild
-  registeredAt: { type: Number, default: Date.now() },
-
+  id: {
+    type: String,
```
Comment (davidzwa) [inline src/database/models/guild.js:5]: This `id` prop although not introduced in this PR might be confusing with the autogenerated `_id` prop which can also be accessed using `id` getter sometimes. The type of `id` is ObjectID which is not compatible with String.

I leave it up to you to determine relevance 👍🏼

File: src/tools/botPermissions.js [hunk 3]
```diff
@@ -45,7 +45,7 @@ export class botPermissions {
   }
 
   checkBotPerms(interaction) {
-    if (this.botPerms.length && !interaction.guild.me.permissions.has(this.botPerms)) {
+    if (this.botPerms.length && !interaction.guild.members.me.permissions.has(this.botPerms)) {
```
Comment (davidzwa) [inline src/tools/botPermissions.js:48]: I get a bit of shivers when seeing the depth of `interaction.guild.members.me.permissions`. But I guess the risk of it breaking on `undefined` errors is very low right?

**Review by davidzwa — APPROVED** [review]
Comments are just for slight eval. LGTM. Didnt compile or test.

Best of luck!

File: src/tools/botPermissions.js (hunk 3 above)
Comment (mkevenaar) [reply src/tools/botPermissions.js:48]: Really low, there would be something really wrong with the Discord API or Discord.JS

**Pull request merged**